src/core/server/app/middleware/passport/strategies/jwt.ts

```diff
@@ -83,17 +95,29 @@ export async function verifyAndRetrieveUser(
   const header: StandardHeader = decoded.header;
   const token: Token = decoded.payload;
 
+  let validationErrors = "";
   try {
     // Try to verify the token.
     for (const verifier of verifiers) {
-      if (verifier.supports(token, tenant)) {
-        return await verifier.verify(
-          tokenString,
-          token,
-          tenant,
-          now,
-          header.kid
-        );
+      // First check that verifier is enabled on the tenant
+      if (verifier.enabled(tenant, token)) {
+        // Then check for token validation errors if verifier is enabled
+        const error = verifier.checkForValidationError(token);
+        if (error) {
+          if (validationErrors.length === 0) {
+            validationErrors += error;
+          } else {
+            validationErrors += " " + error;
+          }
```

Comment:
This is purely my personal preference, I tend to keep lists of error strings and combine them later using `Array.join(", ")` or similar.

```
const validationErrors: string[] = [];
...

if (error) {
   validationErrors.push(error);
}
...

const combinedError = errors.join(", ");
```